diagnostic_tool/diagnostic_script.py

```diff
@@ -0,0 +1,165 @@
+import sys
+import blosc
+import dill
+import numpy as np
+from time import sleep, time
+from functools import partial
+from interbotix_xs_modules.xs_robot.arm import InterbotixManipulatorXS
+
+
+if sys.argv[1] == "deploy":
+    DEPLOY_MODE = True
+else:
+    DEPLOY_MODE = False
+
+
+class WX250sManipulator:
+    def __init__(self):
+        self.bot = InterbotixManipulatorXS("wx250s", "arm", "gripper")
+        self.arm = self.bot.arm
+        self.joint_names = self.arm.group_info.joint_names
+        self.num_joints = len(self.joint_names)
+
+    def get_gripper_tf(self):
+        return self.arm.T_sb
+
+    def get_joint_positions(self):
+        return self.arm.core.joint_states.position[:6]
+
+    # def set_joint_positions(self, joint_positions, moving_time):
+    #     pass
+
+    def create_right_angle(self, moving_time=3.0):
+        self.arm.set_joint_positions([0]*self.num_joints, moving_time=moving_time)
+
+    def point_straight_cartesian(self, moving_time=3.0):
+        self.create_right_angle(moving_time)
+
+        self.arm.set_ee_cartesian_trajectory(x=-0.1, z=-0.2, moving_time=moving_time)
+        print(self.get_joint_positions())
+
+    def point_straight(self, moving_time=3.0):
+        self.create_right_angle(moving_time)
+        if DEPLOY_MODE: sleep(0.5)
+
+        lower_jp = [0.0, -0.00020950394840690834, 0.9273922091469897, 0.0, -0.9271827051985841, 0.0]
+        self.arm.set_joint_positions(lower_jp, moving_time=moving_time)
+        if DEPLOY_MODE: sleep(0.5)
+
+    def compute_joint_trajectories(self, way_points, moving_time=4.0):
+        joint_trajectories = []
+
+        for i, (dx, dy, dz) in enumerate(way_points):
+            print("Computing joint positions for path {}".format(i+1))
+            jt = self.arm.set_ee_cartesian_trajectory(x=dx, y=dy, z=dz,
+                                                      moving_time=moving_time,
+                                                      return_joint_trajectory=True)
+            joint_trajectories.append(jt)
+
+            """ This one doesn't block """
+            # traj = [{0.2 * i: list(point)} for i, point in enumerate(joint_traj)]
+            # self.bot.core.robot_write_trajectory('group', 'arm', 'position', traj)
+            """ This one doesn't work. Test on real robot. """
+            # for jp in joint_traj:
+            #     self.arm.set_joint_positions(jp, moving_time=period)
+            """ No Cartesian ee interpolation. Simply get to endpoint """
+            # p = self.arm.get_ee_pose()
+            # p[0, -1] += dx
+            # p[1, -1] += dy
+            # p[2, -1] += dz
+            # self.arm.set_ee_pose_matrix(p, moving_time=moving_time)
+
+            self.arm.set_ee_cartesian_trajectory(x=dx, y=dy, z=dz, moving_time=moving_time)
```

Comment:
So I assume with `return_joint_trajectory`, it does not actually execute it. But without that argument, it will block and execute the trajectory?

Reply:
This is a dev-file and is no longer exists.